### PR TITLE
pipelines: Follow up on connecting to KFP in multi-user mode.

### PR DESCRIPTION
**Pull request by zijianjoy**
To address the comments in https://github.com/kubeflow/website/pull/2905:

* Process about process of AuthN and AuthZ
* Cross-namespace section title
* ServiceAccountToken explanation.
* Overall doc structure.

cc @bartgras @shuesc1

### CODE REVIEW

File: content/en/docs/components/pipelines/sdk/connect-api.md [hunk 1]
```diff
@@ -6,17 +6,18 @@ weight = 25
 
 This guide demonstrates how to connect to Kubeflow Pipelines using [the Kubeflow Pipelines SDK client](/docs/components/pipelines/sdk/sdk-overview/), and how to [configure the SDK client using environment variables](#configure-sdk-client-by-environment-variables).
 
+
+The Kubeflow Pipelines REST API is available at the same endpoint as the Kubeflow Pipelines user interface (UI).
+The SDK client can send requests to this endpoint to upload pipelines, create pipeline runs, schedule recurring runs and more.
```
Comment (shuesc1) [inline content/en/docs/components/pipelines/sdk/connect-api.md:11]: ..., schedule recurring runs, and more.

File: content/en/docs/components/pipelines/sdk/connect-api.md [hunk 2]
```diff
@@ -204,13 +205,12 @@ Cross-namespace access can be achieved in two ways:
...
   textbox he adds email address associated with the `namespace-1`.
 
-##### How Multi-User mode in-cluster authentication works
+#### How Multi-User mode in-cluster authentication works
 
-Authentication uses ServiceAccountToken 
-[projection](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection). 
-Simply put, the token is first being injected into a Pod (e.g. Jupyter notebook's). 
-Then Kubeflow Pipelines SDK uses this token to authorize against Kubeflow Pipelines API.
+When using Kubeflow Pipelines SDK in the same cluster, it authenticates as default-editor in your namespace using ServiceAccountToken
```
Comment (shuesc1) [inline content/en/docs/components/pipelines/sdk/connect-api.md:210]: "When using Kubeflow Pipelines SDK in the same cluster, [it] authenticates as default-editor.."

Could we be more explicit about the situation and what "it" refers to? For example:

"When multiple users employ the Kubeflow Pipelines SDK in the same cluster, the [service/SDK] authenticates the [namespace/role/user?] as default editor..."

File: content/en/docs/components/pipelines/sdk/connect-api.md [hunk 3]
```diff
@@ -204,13 +205,12 @@ Cross-namespace access can be achieved in two ways:
...
   textbox he adds email address associated with the `namespace-1`.
 
-##### How Multi-User mode in-cluster authentication works
+#### How Multi-User mode in-cluster authentication works
 
-Authentication uses ServiceAccountToken 
-[projection](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection). 
-Simply put, the token is first being injected into a Pod (e.g. Jupyter notebook's). 
-Then Kubeflow Pipelines SDK uses this token to authorize against Kubeflow Pipelines API.
+When calling Kubeflow Pipelines API in the same cluster, Kubeflow Pipelines SDK authenticates itself as `default-editor` in your namespace using ServiceAccountToken
```
Comment (Bobgy) [inline content/en/docs/components/pipelines/sdk/connect-api.md:210]: nit: instead of `default-editor`, it's more accurate to say "your Pod's service account" (or any other words with the same meaning), because the example is using `default-editor`, but people can configure auth with any other service accounts too.
